### PR TITLE
Issue #2568: Add support for "trusted_host_patterns".

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/2568

### CODE REVIEW

**Comment by klonos** [comment]
...refreshing the sandbox.

File: core/includes/bootstrap.inc [hunk 1]
```diff
@@ -4331,3 +4335,28 @@ function backdrop_clear_opcode_cache($filepath) {
     @apc_delete_file($filepath);
   }
 }
+
+function backdrop_check_trusted_hosts($host) {
```
Comment (quicksketch) [inline core/includes/bootstrap.inc:4350]: We need documentation added here for this new function, including a `@since` line.

File: settings.php [hunk 2]
```diff
@@ -313,6 +313,47 @@
...
+ *   '^.+\.example\.com$',
+ *   '^example\.org',
+ *   '^.+\.example\.org',
+ * );
+ * @endcode
+ *
+ * @see backdrop_valid_http_host()
+ * @see backdrop_check_trusted_hosts()
+ * @see system_requirements()
+*/
```
Comment (quicksketch) [inline settings.php:355]: Could we add a commented-out example (I know there are also examples in the docblock above).
```
// $settings['trusted_host_patterns'] = array('^www\.example\.com$');
```

Comment (klonos) [reply settings.php:355]: Sure, done 👍

File: core/includes/bootstrap.inc (hunk 1 above)
Comment (klonos) [reply core/includes/bootstrap.inc:4350]: Sure, will that do?

Comment (klonos) [reply core/includes/bootstrap.inc:4350]: ...oh 🤦‍♂ ...forgot to add `@since`

**Comment by klonos** [comment]
Closing in favor of https://github.com/backdrop/backdrop/pull/2835

**Pull request closed without merging**